lib/whatsapp_sdk/resource/interactive_action.rb

```diff
@@ -6,36 +6,120 @@ module Resource
     class InteractiveAction
       extend T::Sig
 
+      class Type < T::Enum
+        extend T::Sig
+
+        enums do
+          ListMessage = new("list_message")
+          ReplyButton = new("reply_button")
+        end
+      end
+
+      # Returns the type of interactive action you want to send.
+      #
+      # @returns type [Type]. Supported Options are list_message and reply_button.
+      sig { returns(Type) }
+      attr_accessor :type
+
       # Returns the buttons of the Action. For reply_button type, it's required.
       #
-      # @returns buttons [Array<InteractiveActionButton>] .
-      sig do
-       returns(T::Array[InteractiveActionButton])
-      end
+      # @returns buttons [Array<InteractiveActionReplyButton>] .
+      sig { returns(T::Array[InteractiveActionReplyButton]) }
       attr_accessor :buttons
 
-      # TODO: attr_accessor :button
-      # TODO: attr_accessor :sections
+      # Returns the button of the Action. For list_message type, it's required.
+      #
+      # @returns button [String] .
+      sig { returns(String) }
+      attr_accessor :button
```

Comment:
why is it a `string`?

Reply:
This represents the `button` key of `interactive` object which is a text field only, as in, it does not have a `{ text: 'button' }` or `{ type: 'text', text: 'button' }` structure.